app/assets/javascripts/my_modules/samples_widget.js

```diff
@@ -0,0 +1,56 @@
+//= require datatables
+
+table = $("#samples").DataTable({
+    order: [[2, "desc"]],
+    dom: "RB<'row'<'col-sm-9-custom toolbar'l><'col-sm-3-custom'f>>ti",
+    stateSave: false,
+    buttons: [],
+    bPaginate: false,
+    searching: false,
+    processing: true,
+    serverSide: true,
+    ajax: {
+        url: $("#samples").data("source"),
+        global: false,
+        type: "POST"
+    },
+    colReorder: {
+        fixedColumnsLeft: 1000000 // Disable reordering
+    },
+    columnDefs: [{
+        targets: 0,
+        searchable: false,
+        orderable: false,
+        className: "dt-body-center",
+        sWidth: "1%",
+        render: function (data, type, full, meta){
+            return "<input type='checkbox'>";
+        }
+    }, {
+        targets: 1,
+        searchable: false,
+        orderable: true,
+        sWidth: "1%"
+    }],
+    columns: (function() {
```

Comment:
I suggest you make a same `columnDefs` for all columns (e.g. `targets: [0, 1, 2, 3, 4, 5, 6]`), and change the `columns` definition to just show the ones defined in HTML.